src/main/scala/nl/biopet/tools/bamstats/generate/Generate.scala

```diff
@@ -86,48 +99,40 @@ object Generate extends ToolCommand[Args] {
     logger.info("Done")
   }
 
-  def extractStats(bamFile: File,
-                   bedFile: Option[File],
-                   sequenceDict: SAMSequenceDictionary,
-                   scatterMode: Boolean = false,
-                   onlyUnmapped: Boolean = false): GroupStats = {
+  def extractStats(bamFile: File): GroupStats = {
     val samReader: SamReader =
       SamReaderFactory.makeDefault().open(bamFile)
     val stats = GroupStats()
+    samReader.iterator().foreach(stats.loadRecord)
+    samReader.close()
+    stats
+  }
 
-    bedFile match {
-      case Some(bed: File) if onlyUnmapped =>
-        throw new IllegalArgumentException(
-          "Cannot extract stats from regions and unmapped regions at the same time")
-      // When a BED file is specified extract regions stats
-      case Some(bed: File) if !onlyUnmapped =>
-        val regions =
-          BedRecordList
-            .fromFile(bed)
-            .combineOverlap
-            .validateContigs(sequenceDict)
-        regions.allRecords.foreach { bedRecord =>
-          val samRecordIterator: SAMRecordIterator =
-            samReader.query(bedRecord.chr,
-                            bedRecord.start,
-                            bedRecord.end,
-                            false)
-          samRecordIterator.foreach { samRecord =>
-            // Read based stats
-            // If scatterMode is false, continue.
-            // If scatterMode is true, determine whether the alignment start is within the region.
-            if (!scatterMode || samRecord.getAlignmentStart > bedRecord.start && samRecord.getAlignmentStart <= bedRecord.end) {
-              stats.loadRecord(samRecord)
-            }
-          }
-        }
-      // If onlyUnmapped is set only get stats for unmapped reads.
-      case None if onlyUnmapped =>
-        samReader.queryUnmapped().foreach(stats.loadRecord)
-      // If no regions or unmapped files are set determine stats for all records
-      case None if !onlyUnmapped =>
-        val records: Iterator[SAMRecord] = samReader.iterator().toIterator
-        records.foreach(stats.loadRecord)
+  def extractStatsUnmappedReads(bamFile: File): GroupStats = {
+    val samReader: SamReader =
+      SamReaderFactory.makeDefault().open(bamFile)
+    val stats = GroupStats()
+    samReader.queryUnmapped().foreach(stats.loadRecord)
+    samReader.close()
+    stats
+  }
+
+  def extractStatsRegion(bamFile: File,
```

Comment:
This method is fine. But for performance  is also good to have a method like this:
```scala
def extractStatsRegion(reader: SamReader,
                       region: BedRecord, 
                       scatterMode: Boolean = false): GroupStats = ???
```

This way you can do multiple regions with 1 SamReader, opening is slow when having small regions like exons.